mo_net/model/layer/output.py

```
@@ -248,3 +261,320 @@ def output_dimensions(self) -> Dimensions:
+        for word_id in range(self.vocab_size):
+            node_indices, directions = self.tree.get_path(word_id)
+
+            if len(node_indices) == 0:
+                # Edge case: single word vocabulary
+                log_probs = log_probs.at[:, word_id].set(0.0)
+                continue
+
+            # Compute log P(word|context) = sum of log P(decision_i) along path
+            word_log_prob = jnp.zeros(batch_size)
+
+            for node_idx, go_left in zip(node_indices, directions):
```

Ruff (0.14.14)

[warning] `zip()` without an explicit `strict=` parameter

Add explicit value for parameter `strict=`

(B905)